src/metabase/query_processor/expand.clj

```diff
@@ -393,21 +393,47 @@
 (s/defn ^:ql ^:always-validate expressions
   "Top-level clause. Add additional calculated fields to a query."
   {:added "0.17.0"}
-  [query, m :- {s/Keyword Expression}]
+  [query, m :- {s/Keyword i/AnyExpression}]
   (assoc query :expressions m))
 
-(s/defn ^:private ^:always-validate expression-fn :- Expression
+(s/defn ^:private ^:always-validate expression-fn :- ArithmeticExpression
   [k :- s/Keyword, & args]
-  (i/map->Expression {:operator k, :args (vec (for [arg args]
-                                                (if (number? arg)
-                                                  (float arg) ; convert args to floats so things like 5 / 10 -> 0.5 instead of 0
-                                                  arg)))}))
+  (i/map->ArithmeticExpression {:operator k, :args (vec (for [arg args]
+                                                          (if (number? arg)
+                                                            (float arg) ; convert args to floats so things like 5 / 10 -> 0.5 instead of 0
+                                                            arg)))}))
 
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} + "Arithmetic addition function."       (partial expression-fn :+))
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} - "Arithmetic subtraction function."    (partial expression-fn :-))
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} * "Arithmetic multiplication function." (partial expression-fn :*))
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} / "Arithmetic division function."       (partial expression-fn :/))
 
+(s/defn ^:ql sql-expression
+  "Expression representing arbitrary SQL."
+  [sql-expr :- su/NonBlankString]
+  (i/map->SQLExpression {:expr sql-expr}))
+
+(s/defn ^:ql remap-expression
+  "Expression that projects out alternate representations of values in
+  `COLUMN`. `COLUMN` values are the keys in `MAPPING`, with the value
+  found in `MAPPING` as the result of the expression."
+  [column :- i/FieldPlaceholderOrExpressionRef
+   mapping :- {s/Str s/Str}]
+  (i/map->RemapExpression {:column column
+                           :mapping (reduce-kv (fn [acc k v]
+                                                 (assoc acc (Integer/parseInt k) v))
```

Comment:
This assumes that the data coming in is integers (which I'm encoding as a string after a discussion with @camsaul). Although it seems like integers are definitely the most common usage of this, I've also seen single character values used as a type of enum as well. Are integers a good place to start here? Or should we make the interchange here support any valid type?